src/main/java/com/cloudbees/jenkins/plugins/bitbucket/client/BitbucketCloudApiClient.java

```diff
@@ -106,6 +110,10 @@
         connectionManager.getParams().setDefaultMaxConnectionsPerHost(20);
         connectionManager.getParams().setMaxTotalConnections(22);
     }
+    private static Cache<String, BitbucketTeam> cachedTeam = new Cache(6, TimeUnit.HOURS);
+    private static Cache<UriTemplate, List<BitbucketCloudRepository>> cachedRepositories = new Cache(3, TimeUnit.HOURS);
```

Comment:
is it wise to use the template as the key?

Reply:
yeah... template seems a very dodgy key to use as it will be a new instance for each new `BitbucketCloudApiClient` instance and https://github.com/damnhandy/Handy-URI-Templates/blob/master/src/main/java/com/damnhandy/uri/template/UriTemplate.java#L68 does not override `hashCode` or `equals` so every new `BitbucketCloudApiClient` will result in a new listing of all repositories (i.e. defeating the purpose of the cache in the first place)

I think you need a synthetic key for both caches that reflects:

* The owner
* The repository
* The username from the credentials (will need to revisit for api tokens)

You can either have a data class that reflects the correct equality and hashcode such that two equivalent instances are equal or you can just munge it all into a string, e.g.

`String cacheKey = owner + "::" + repositoryName + "::" + creds.getUsername(); // relying on '::' not being valid for repoName or owner name in bitbucket.org`

Reply:
sorry, did not look so far.
Of course, will change

Reply:
@stephenc why do you want to add repository and username to the cache key ?
Since the template only use the owner and, if set, the role, I would only use them, no ?

Reply:
@bguerin isn't owner good enough as the key for the owners repos? though you might have to consider any filter upon repositories, and changes to the configuration. Don't know if a change to the current config of the Jenkins Team job configuration will clear the cache.

Reply:
Think same. I will use owner, and concat role if defined. Ok for you ?

Reply:
For me, yes. Think Steven might have misunderstood where the cached repositories were being used.

Reply:
for cachedRepositories, you need owner and credentials used.

Different credentials may have different permissions on the available repositories, so the answer you get in asking for a list of repositories may depend on the credentials you ask with. If you don't include the credentials username then we either leak details of repositories that should be hidden to users that shouldn't see them or we hide repositories that should be seen from users that should see them depending on which credentials populated the cache.

I think cachedTeam needs the repo included in the cache key but I could be wrong on that one

Reply:
done